cli/src/program_v4.rs

```diff
@@ -677,11 +677,6 @@ pub fn process_deploy_program(
         if !program_account_exists {
             return Err("Program account does not exist".into());
         }
-    } else {
```

Comment:
What's the significance of removing this check? Also, what does the partial deployment means?

Reply:
It means one can update only a chunk of a program account, and does not have to always upload the entire file even if most stays the same. It is kinda splitting hairs if that should be counted as the same deployment or a redeployment. Maybe it can be done while keeping the check.